01_classify_stratified/run_cancer_type_classification.py

```diff
@@ -0,0 +1,165 @@
+    sample_info_df = du.load_sample_info(args.verbose)
+    tcga_cancer_types = list(np.unique(sample_info_df.cancer_type))
+
+    if args.cancer_types is None:
+        args.cancer_types = tcga_cancer_types
+    else:
+        not_in_tcga = set(args.cancer_types) - set(tcga_cancer_types)
+        if len(not_in_tcga) > 0:
+            p.error('some cancer types not present in TCGA: {}'.format(
+                ' '.join(not_in_tcga)))
+
+    return args, sample_info_df
+
+
+if __name__ == '__main__':
+
+    # process command line arguments
+    args, sample_info_df = process_args()
+
+    # create results dir if it doesn't exist
+    args.results_dir.mkdir(parents=True, exist_ok=True)
+
+    # create empty log file if it doesn't exist
+    log_columns = [
+        'cancer_type',
+        'training_data',
+        'shuffle_labels',
+        'skip_reason'
+    ]
+    if args.log_file.exists() and args.log_file.is_file():
+        log_df = pd.read_csv(args.log_file, sep='\t')
+    else:
+        log_df = pd.DataFrame(columns=log_columns)
+        log_df.to_csv(args.log_file, sep='\t')
+
+    tcga_data = TCGADataModel(seed=args.seed,
+                              subset_mad_genes=args.subset_mad_genes,
+                              training_data=args.training_data,
+                              verbose=args.verbose,
+                              debug=args.debug)
+
+    # we want to run mutation prediction experiments:
+    # - for true labels and shuffled labels
+    #   (shuffled labels acts as our lower baseline)
+    # - for all cancer types in the given list of TCGA cancers
+    for shuffle_labels in (False, True):
+
+        print('shuffle_labels: {}'.format(shuffle_labels))
+
+        progress = tqdm(args.cancer_types,
+                        total=len(args.cancer_types),
+                        ncols=100,
+                        file=sys.stdout)
+
+        for cancer_type in progress:
+            cancer_type_log_df = None
+            progress.set_description('cancer type: {}'.format(cancer_type))
+
+            try:
+                cancer_type_dir = fu.make_cancer_type_dir(args.results_dir, cancer_type)
+                check_file = fu.check_cancer_type_file(cancer_type_dir,
+                                                       cancer_type,
+                                                       args.training_data,
+                                                       shuffle_labels,
+                                                       args.seed)
+                tcga_data.process_data_for_cancer_type(cancer_type,
+                                                       cancer_type_dir,
+                                                       shuffle_labels=shuffle_labels)
+            except ResultsFileExistsError:
+                # this happens if cross-validation for this cancer type has
+                # already been run (i.e. the results file already exists)
```

Comment:
I do something similar to what you just described: I try to put the important parameters in the file names where I store results. I don't know either if it is a great solution (probably not), but I think the important part is to avoid a mismatch between the results in disk and what you think there is there. Usually, at the end of all refactoring and reviews, I just run everything again just in case.